src/mapping.ts

```diff
@@ -17,16 +19,81 @@ const NON_CIRCULATING_GCHAIN = [
   // TODO - add GnosisDAO Vesting contract Gchain
 ];
 
+export function loadOrCreateHolder(address: Address): Holder {
+  let holder = Holder.load(address.toHex());
+  if (!holder) {
+    holder = new Holder(address.toHex());
+  }
+  if (address != Address.zero()) {
+    // Skip creation of burn address holder.
+    holder.save();
+  }
+  return holder;
+}
+
+export function saveNonZero(holder: Holder): void {
+  if (holder.id != Address.zero().toHex()) {
+    holder.save();
+  }
+  // do not save updates to null address balance.
+}
+
+export function loadOrCreateSupply(): Supply {
+  let supply = Supply.load(COW_TOKEN);
+  if (!supply) {
+    // This will only ever happen once!
+    supply = new Supply(COW_TOKEN);
+  }
+  supply.save();
+  return supply;
+}
+
+export function updateSupply(
+  from: Address,
+  to: Address,
+  amount: BigInt,
+  excluded: Address[]
+): void {
+  const supply = loadOrCreateSupply();
+  if (from == Address.zero()) {
+    log.info("Token Minted {}", [amount.toString()]);
+    supply.total = supply.total.plus(amount);
+    supply.circulating = supply.circulating.plus(amount);
+  }
+  if (to == Address.zero()) {
+    log.info("Token Burned {}", [amount.toString()]);
+    supply.total = supply.total.minus(amount);
+    supply.circulating = supply.circulating.minus(amount);
+  }
+
+  if (excluded.includes(from)) {
```

Comment:
How are you planning to handle the transfers that we sent from cowDAO to the bridging contract? If they are vested on Gnosis chain, they should not increase the supply, right

Reply:
Yes. this is a very good point. I was meaning to address this. However it seems like we are going to go with only a single mainnet subgraph (for simplicity). in this case I woudl forward us to the mainnet only subgraph PR #6

#

Reply:
If we do this, then we may as well consider the bridged funds as "circulating".